exercises/concept/arcade-high-score/src/test/java/ArcadeHighScoreTest.java

```diff
@@ -0,0 +1,90 @@
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+public class ArcadeHighScoreTest {
+
+    private final ArcadeHighScore arcadeHighScore = new ArcadeHighScore();
+
+
+    @Test
+    @Tag("task:1")
+    @DisplayName("Define a map to store the high scores")
+    public void defineMap() {
+        assertNotNull(arcadeHighScore.defineMap());
+    }
+
+    @Test
+    @Tag("task:2")
+    @DisplayName("Add a player to the high scores map")
+    public void addPlayer() {
+        String name = "David James";
+        Integer score = 67;
+
+        arcadeHighScore.addPlayer(name, score);
+        assertThat(arcadeHighScore.highScores).containsEntry(name, score);
+    }
+
+    @Test
+    @Tag("task:3")
+    @DisplayName("Remove a player from the high scores map")
+    public void removePlayer() {
+        String name = "David James";
+        Integer score = 67;
+
+        arcadeHighScore.addPlayer(name, score);
+        assertThat(arcadeHighScore.highScores).containsEntry(name, score);
+
+        arcadeHighScore.removePlayer(name);
+        assertThat(arcadeHighScore.highScores).isEmpty();
+    }
+
+    @Test
+    @Tag("task:4")
+    @DisplayName("Reset a players score to 0")
+    public void resetScore() {
+        String name = "David James";
+        Integer score = 67;
+
+        arcadeHighScore.addPlayer(name, score);
+        assertThat(arcadeHighScore.highScores).containsEntry(name, score);
+
+        arcadeHighScore.resetScore("David James");
+        assertThat(arcadeHighScore.highScores).containsEntry(name, 0);
+    }
+
+    @Test
+    @Tag("task:5")
+    @DisplayName("Update a players score")
+    public void updateScore() {
```

Comment:
I think we need to add a second test for this task that verifies the requirement when `updateScore` is called with a player that doesn't exist in the map yet.